rockcraft/oci.py

```diff
@@ -215,34 +219,86 @@ def to_oci_archive(self, tag: str, filename: str) -> None:
         _copy_image(f"oci:{str(src_path)}", f"oci-archive:{filename}:{tag}")
 
-    def set_entrypoint(self, entrypoint: List[str]) -> None:
-        """Set the OCI image entrypoint.
-
-        :param entrypoint: The default list of arguments to use as the command to
-            execute when the container starts.
-        """
+    def set_entrypoint(self) -> None:
+        """Set the OCI image entrypoint. It is always Pebble and CMD is null."""
         emit.progress("Configuring entrypoint...")
         image_path = self.path / self.image_name
+        entrypoint = [f"/{self._PEBBLE_BINARY_PATH}", "enter"]
         params = ["--clear=config.entrypoint"]
         for entry in entrypoint:
             params.extend(["--config.entrypoint", entry])
         _config_image(image_path, params)
+        # Clear the CMD
+        _config_image(image_path, ["--clear=config.cmd"])
         emit.progress(f"Entrypoint set to {entrypoint}", permanent=True)
 
-    def set_cmd(self, cmd: List[str]) -> None:
-        """Set the OCI image default command parameters.
-
-        :param cmd: The default arguments to the entrypoint of the container. If
-            entrypoint is not defined, the first entry of cmd is the executable to
-            run when the container starts.
+    def set_pebble_services(
+        self,
+        services: Dict[str, Any],
+        name: str,
+        tag: str,
+        summary: str,
+        description: str,
+        base_layer_dir: Path,
+    ) -> None:
+        """Write the provided services into a Pebble layer in the filesystem.
+
+        :param services: The Pebble services to be written into the Pebble
+        :param name: The name of the ROCK
+        :param tag: The ROCK's image tag
+        :param summary: The summary for the Pebble layer
+        :param description: The description for the Pebble layer
+        :param base_layer_dir: Path to the base layer's root filesystem
         """
-        emit.progress("Configuring cmd...")
-        image_path = self.path / self.image_name
-        params = ["--clear=config.cmd"]
-        for entry in cmd:
-            params.extend(["--config.cmd", entry])
-        _config_image(image_path, params)
-        emit.progress(f"Cmd set to {cmd}", permanent=True)
+        # pylint: disable=too-many-locals,too-many-arguments
+        service_names = list(services.keys())
+        emit.progress(f"Configuring Pebble services {', '.join(service_names)}")
+        pebble_layer_content = {
+            "summary": summary,
+            "description": description,
+            "services": services,
+        }
+
+        # To infer the right filename for this Pebble layer, we should check
+        # if there are existing Pebble layers in the base root filesystem,
+        # and increment on that.
+        # NOTE: the layer's filename prefix will always be "001-" when using
+        # "bare" and "ubuntu" bases
+        pebble_layers_path_in_base = f"{base_layer_dir}/{self._PEBBLE_LAYERS_PATH}"
```

Comment:
I feel like this method is doing too much work and only some of it is related to the OCI layer that we're adding. What do you think about the following suggestion:
Add a `pebble` module like the other comment (from me, heh) suggested. Then add a function like `add_pebble_services(target_dir, services, ...)` that will do the meat of this method. Then this method can be something like:

```python
def set_pebble_services(...):
    with tempfile.TemporaryDirectory() as td:
        tmpfs = Path(td)
        pebble.add_pebble_services(tmpfs, services, base_layer_dir)
        self.add_layer(tag, tmpfs)
```

Reply:
done